core/processor/src/main/java/io/quarkus/annotation/processor/extension/ExtensionBuildProcessor.java

```diff
@@ -67,7 +68,23 @@ public void process(Set<? extends TypeElement> annotations, RoundEnvironment rou
     public void finalizeProcessing() {
         validateAnnotationUsage();
 
-        utils.filer().write(Outputs.META_INF_QUARKUS_BUILD_STEPS, buildSteps);
+        /*
+         * During an incremental compilation (i.e. while developing extensions in Intellij IDEA)
+         * the Annotation Processor API will include only changed classes
+         * creating a subset of processors that are not enough to run a quarkus app
+         * By assuming a full compilation was made initially, all the processors are included inside the
+         * META-INF/quarkus-build-steps.list file
+         * So by reading it we can ensure that all the build steps are included.
+         * See
+         * https://youtrack.jetbrains.com/issue/IJPL-196660/During-an-incremental-build-getElementsAnnotatedWith-doesnt-include-
+         * all-the-elements-but-only-the-one-recompiled
+         */
+        List<String> allProcessors = utils.filer().read(Outputs.META_INF_QUARKUS_BUILD_STEPS);
+
+        Set<String> allBuildSteps = new HashSet<String>(allProcessors);
```

Comment:
```suggestion
        Set<String> allBuildSteps = new TreeSet<>(allProcessors);
```
(currently working on build reproducibility :))